sklearn/utils/estimator_checks.py

```diff
@@ -1197,6 +1198,36 @@ def check_classifiers_one_label(name, classifier_orig):
             raise exc
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_classifiers_one_label_sample_weights(name, classifier_orig):
+    # check that classifiers accepting sample_weight fit fine or
+    # throws an ValueError if the problem is reduce to one class.
+    error_fit = ("Classifier can't train when only one class is present "
```

Comment:
less that two classes are present?

Reply:
https://github.com/scikit-learn/scikit-learn/blob/c30c503951706a648bad0f2587c3c06dc99a9310/sklearn/utils/estimator_checks.py#L1167-L1168

I must admit that I just did a copy/paste from the `check_classifiers_one_label`, but I can change it.

Reply:
I just thought of this change as a possibility. But I guess we can wait for a code dev to review.